manifests/apache/vhost.pp

```diff
@@ -87,7 +87,8 @@
   Optional[String] $ldap_bind_dn            = undef,
   Optional[String] $ldap_bind_password      = undef,
   Optional[String] $ldap_url                = undef,
-  Optional[String] $ldap_bind_authoritative = undef
+  Optional[String] $ldap_bind_authoritative = undef,
+  Optional[Hash] $custom_apache_parameters  = {},
```

Comment:
@ekohl i can change this if you want

Reply:
@ekohl why using `{}` instead of undef ? When using `{}` then every optional string would IMHO need `''` instead of undef to be consistent. But then the parameters aren't  `Optional` anymore because IMHO the empty string or the empty hash is already a value.

Reply:
@vinzent because it gets used in a `* => $custom_apache_parameters` notation - you can't give `undef` as an argument to `*` so this would cause compilation failure - defaulting to `{}` is a lot cleaner than having to wrap things in conditionals... I do agree however that the use of `Optional` is not needed